lua/astronvim/utils/status/heirline.lua

```diff
@@ -47,7 +47,7 @@ M.make_buflist = function(component)
           right = "tabline_bg",
         }
       end,
-      { -- bufferlist
+      {              -- bufferlist
         init = function(self) self.tab_type = M.tab_type(self) end,
```

Comment:
This is where you should be calculating `self.bufnr` for each tab in the tabline which corresponds to each vim-tab and therefore is where the bufnr should be calculated a single time. It shouldn't be populated as a function through each provider.